protocols/api/src/main/java/org/apache/james/protocols/api/ProtocolConfigurationImpl.java

```diff
@@ -44,7 +44,14 @@
         }
         DEFAULT_HELLO_NAME = hName;
     }
-    
+
+    private ProtocolConfigurationImpl() {
+    }
+
+    protected ProtocolConfigurationImpl(String softwareName) {
+        this.softwareName = softwareName;
+    }

```

Comment:
You can set them `final` and use `Optional`

Reply:
that's funny, I explored the fields possible values and many of them are completely useless